specification/eventgrid/data-plane/Microsoft.Web/stable/2018-01-01/Web.json

```diff
@@ -0,0 +1,679 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-01-01",
+    "title": "Schema of Azure App Service events published to Azure Event Grid",
+    "description": "Describes the schema of the Azure App Service events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+  },
+  "paths": {},
+  "definitions": {
+    "WebAppDeletedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for an Microsoft.Web.AppDeleted event.",
+      "type": "object",
+      "properties": {
+        "appEventTypeDetail": {
+          "description": "Detail of action on the app",
+          "type": "object",
+          "properties": {
+            "action": {
+              "description": "Type of action of the operation.",
+              "type": "string"
+            }
+          }
+        },
+        "name": {
+          "description": "name of the web site that had this event.",
+          "type": "string"
+        },
+        "clientRequestId": {
+          "description": "The client request id generated by the app service for the site API operation that triggered this event.",
+          "type": "string"
+        },
+        "correlationRequestId": {
+          "description": "The correlation request id generated by the app service for the site API operation that triggered this event.",
+          "type": "string"
+        },
+        "requestId": {
+          "description": "The request id generated by the app service for the site API operation that triggered this event.",
+          "type": "string"
+        },
+        "address": {
+          "description": "HTTP request URL of this operation.",
+          "type": "string"
+        },
+        "verb": {
+          "description": "HTTP verb of this operation.",
+          "type": "string"
+        }
+      }
+    },
+    "WebAppUpdatedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for an Microsoft.Web.AppUpdated event.",
+      "type": "object",
+      "properties": {
+        "appEventTypeDetail": {
+          "description": "Detail of action on the app",
+          "type": "object",
+          "properties": {
+            "action": {
```

Comment:
It's the second occurrence of a anonymous class that contains "action", SDK will auto-create a different class with a different name, you might benefits by doing an explicit "OperationActionType" class that contains "action", and `ref` it each time you need it.